File: fetch.php
<?php

$json = json_decode(file_get_contents(__DIR__ . '/cli/config.json'), true);

define('DEFAULT_HRS', 72);

$hrs = DEFAULT_HRS;
if (array_key_exists("hrs", $_GET)) {
  $hrs = $_GET["hrs"];
}

try {
  $db = new mysqli($json['db_host'], $json['db_user'], $json['db_pass'], $json['db_name']);
  if ($db->connect_error) {
    throw new Exception($db->connect_error);
  }
  if ($stmt = $db->prepare("SELECT * from data where timestamp>=DATE_SUB(NOW(), INTERVAL ? HOUR) order by timestamp")) {
    $stmt->bind_param("i", $hrs);
    $stmt->execute();
    $stmt->bind_result($timestamp, $heating, $target, $current, $humidity, $updated);
    header("Content-type: text/tab-separated-values");
    print "timestamp\theating\ttarget\tcurrent\thumidity\tupdated\n";
    while ($stmt->fetch()) {
      print implode("\t", array($timestamp, $heating, $target, $current, $humidity, $updated)) . "\n";
    }
    $stmt->close();
  }
  $db->close();
} catch (Exception $e) {
  $errors[] = ("DB connection error! <code>" . $e->getMessage() . "</code>.");
}

?>

File: fetch.py
#!/usr/bin/env python3
import cgi
import json
import os
import sys

SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
sys.path.insert(0, os.path.join(SCRIPT_DIR, "cli", "venv", "lib",
                f"python{sys.version_info.major}.{sys.version_info.minor}",
                "site-packages"))

import mysql.connector

CONFIG_PATH = os.path.join(SCRIPT_DIR, "cli", "config.json")
DEFAULT_HRS = 72

params = cgi.FieldStorage()
try:
    hrs = int(params.getfirst("hrs", DEFAULT_HRS))
except (ValueError, TypeError):
    hrs = DEFAULT_HRS

with open(CONFIG_PATH) as f:
    config = json.load(f)

try:
    conn = mysql.connector.connect(
        host=config["db_host"],
        user=config["db_user"],
        password=config["db_pass"],
        database=config["db_name"],
    )
    cursor = conn.cursor()
    cursor.execute(
        "SELECT * FROM data WHERE timestamp >= DATE_SUB(NOW(), INTERVAL %s HOUR) ORDER BY timestamp",
        (hrs,),
    )

    print("Content-Type: text/tab-separated-values")
    print()
    print("timestamp\theating\ttarget\tcurrent\thumidity\tupdated")
    for row in cursor:
        print("\t".join(str(col) for col in row))

    cursor.close()
    conn.close()
except Exception as e:
    print("Content-Type: text/plain")
    print("Status: 500 Internal Server Error")
    print()
    print(f"DB connection error: {e}")
